File: 496-NextGreaterElementI/496-NextGreaterElementI.java
// Last updated: 8/13/2025, 8:19:43 PM
class Solution {
    public int[] nextGreaterElement(int[] a, int[] b) {
    int[] c = new int [a.length];
    int counter =0;
    int flag;
    for(int i=0;i<a.length;i++){
        flag = 0;
      for(int j=0;j<b.length;j++){
        if(a[i]==b[j]){
          counter = j;
          while(counter<b.length){
            if(b[counter]>b[j]){
              c[i] = b[counter];
              flag = 1;
              break;
            }
            counter++;
          }
          if(flag ==0){
            c[i] = -1;
          }
        }
      }
    }
  return c;
  }
}

File: 496-NextGreaterElementI/496-NextGreaterElementI.py
# Last updated: 8/13/2025, 8:19:37 PM
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        my_stack = []
        i = 0
        hashmap = {}

        while i < len(nums2) : 

            while len(my_stack) > 0 and my_stack[-1][0] < nums2[i] : 
                popped = my_stack.pop(-1)
                popped_element = popped[0]
                popped_index = popped[1]
                hashmap[nums2[popped_index]] = nums2[i]
            
            my_stack.append([nums2[i] , i])
            i+=1
        
        result = []

        for i in range(len(nums1)) : 
            result.append(hashmap.get(nums1[i])) if nums1[i] in hashmap else result.append(-1)
        
        return result
